horovod/common/operations.cc

```diff
@@ -1164,14 +1165,35 @@ void CheckForStalledTensors(HorovodGlobalState& state) {
 //      response from the coordinator. At that point, the tick ends.
 //      If instead of "DONE" they receive "SHUTDOWN", they exit their background
 //      loop.
-void BackgroundThreadLoop(HorovodGlobalState& state) {
+void BackgroundThreadLoop(HorovodGlobalState& state, int mpi_init,
+                          int mpi_finalize) {
   // Initialize MPI. This must happen on the background thread, since not all
   // MPI implementations support being called from multiple threads.
   //
   // We will ask for multiple threads, so other libraries like mpi4py can
   // be used together with Horovod if multi-threaded MPI is installed.
   int provided;
-  MPI_Init_thread(NULL, NULL, MPI_THREAD_MULTIPLE, &provided);
+  int is_mpi_initialized = 0;
+  MPI_Initialized(&is_mpi_initialized);
+
+  // Calls MPI_Init if mpi_init is set otherwise prints warning
+  if (mpi_init) {
+    if (is_mpi_initialized) {
+      std::cerr << " WARNING: MPI has already been initialized, please use "
+                   "hvd.init(mpi_init=False, mpi_finalize=False) if you're "
+                   "using other "
+                   "frameworks that initialize MPI prior to Horovod."
+                << std::endl;
```

Comment:
If this happens, do you see MPI errors due to double finalization?  If so, we should mention it here.

Reply:
Let's do the check here for thread support and if multithreading is not supported, print another warning similar to below that process may crash.